src/main/kotlin/com/openlattice/assembler/AssemblerConnectionManager.kt

```diff
@@ -465,6 +478,7 @@ class AssemblerConnectionManager(
      * Build grant select sql statement for a given table and user with column level security.
      * If properties (columns) are left empty, it will grant select on whole table.
      */
+    @Throws(NoSuchElementException::class)
```

Comment:
Where exactly does this `NoSuchElementException` get thrown? I had thought it came from `stmt.addBatch` 🤔

Reply:
It's coming from the HazelcastPrincipalService.lookupRole -> Optional.get()
But at this point, the principal should exist and should be a Role

Reply:
Shouldn't it be catching these exceptions now though? So theoretically they won't ever get thrown